test/json_converters/enum_converter_test.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:data_class_plugin/public/json_converter.dart';
+import 'package:test/scaffolding.dart';
+import 'package:test/expect.dart';
+import 'dart:convert';
+
+enum Colors { red, green, blue }
+
+class ColorsConverter implements JsonConverter<Colors, String> {
+  const ColorsConverter();
+
+  @override
+  String toJson(Colors value) => value.name;
+
+  @override
+  Colors fromJson(String value) {
+    return Colors.values.firstWhere((Colors c) => c.name == value);
+  }
+}
+
+class MockModel {
+  /// Shorthand constructor
+  MockModel({
+    required this.colors,
+  });
+
+  final Colors colors;
+
+  /// Converts [MockModel] to a [Map] json
+  Map<String, dynamic> toJson() {
+    return <String, dynamic>{
+      'colors': jsonConverterRegistrant.find(Colors).toJson(colors),
+    };
+  }
+
+  /// Creates an instance of [MockModel] from [json]
+  factory MockModel.fromJson(Map<String, dynamic> json) {
+    return MockModel(
+      colors: jsonConverterRegistrant.find(Colors).fromJson(json['colors']) as Colors,
+    );
+  }
+}
+
+void main() {
+  const Colors mockColor = Colors.green;
+  final Map<String, dynamic> mockJson = <String, dynamic>{'colors': mockColor.name};
+
+  group('Test enum json converter', () {
+    final MockModel mockModel = MockModel(colors: mockColor);
+
+    test('throws Exception if not registered', () {
+      expect(
+        () => mockModel.toJson(),
+        throwsA(isA<Exception>()),
+      );
+    });
+
+    test('toJson', () {
```

Comment:
Maybe we need to add method that allows you to remove a converter. At least for testing